lib/core/src/Cardano/Pool/DB.hs

```diff
@@ -223,6 +228,15 @@ data DBLayer m = forall stm. (MonadFail stm, MonadIO stm) => DBLayer
         --    - 'listRetiredPools'.
         --    - 'removePools'.
 
+    , readSettings
+        :: stm Settings
+        -- ^ Get the settings.
+
+    , modSettings
+        :: (Settings -> Settings)
+        -> stm ()
+        -- ^ Modify the settings.
```

Comment:
I would prefer `putSettings`, like is done for the other DBLayer methods. There is no need for a modify-type function when you have read and put (and transactions).